app/src/test/java/org/fossasia/susi/ai/signup/SignUpPresenterTest.kt

```diff
@@ -73,4 +101,80 @@ class SignUpPresenterTest {
         verify(signUpView).showProgress(true)
         verify(iSignUpModel).signUp("email@gmail.com", "P@ssword12", signUpPresenter)
     }
+
+    @Test
+    fun shouldShowErrorOnFailure() {
+        signUpPresenter.onError(Throwable())
```

Comment:
Ideally this should be like 
~~~
when(iSignUpModel.signUp(anyString(),anyString(),any())).then{signUpPresenter.onError(any())}
signUpPresenter.signUp()
~~~
but this doesn't works... that's why I have to call onError directly. Any suggestions??

Reply:
@iamareebjamal please answer... I have to use same pattern in future unit tests also.
open-event-organizer uses observables, so it's somewhat easier to test. I think directly calling onError instead of signUp doesn't makes much difference.

Reply:
also, should I also include tests for minor functions as
~~~
signUpPresenter.cancelSignUP()
verify(signUpModel).cancelSignup()
~~~